sklearn/preprocessing/_encoders.py

```diff
@@ -336,6 +397,44 @@ def _validate_keywords(self):
                 "specified, as both would create categories that are all "
                 "zero.")
 
+        if self.max_categories is not None and self.max_categories <= 1:
+            raise ValueError("max_categories must be greater than 1")
+
+        if isinstance(self.min_frequency, numbers.Integral):
+            if not self.min_frequency >= 1:
+                raise ValueError("min_frequency must be an integer at least "
+                                 "1 or a float in (0.0, 1.0); got the "
+                                 f"integer {self.min_frequency}")
+        else:  # float
+            if not 0.0 < self.min_frequency < 1.0:
```

Comment:
```suggestion
            if not (0.0 < self.min_frequency < 1.0):
```
I would have guessed that `not` has priority over < as in C which doesn't seem to be the case in Python.